packages/transactions/src/fetch.ts

```
@@ -0,0 +1,382 @@
+} & ApiParam): Promise<TxBroadcastResult> {
+  const tx = bytesToHex(txOpt.serialize());
+  const attachment = attachOpt
+    ? typeof attachOpt === 'string'
+      ? attachOpt
+      : bytesToHex(attachOpt)
+    : undefined;
+  const json = attachOpt ? { tx, attachment } : { tx };
+  const options = {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify(json),
+  };
+
+  const api = defaultApiFromNetwork(deriveNetworkFromTx(txOpt), apiOpt);
+  const url = `${api.url}${BROADCAST_PATH}`;
+  const response = await api.fetch(url, options);
+
+  if (!response.ok) {
+    try {
+      return (await response.json()) as TxBroadcastResultRejected;
+    } catch (e) {
+      throw Error('Failed to broadcast transaction (unable to parse node response).', { cause: e });
+    }
+  }
+
+  const text = await response.text();
+  const txid = text.replace(/["]+/g, ''); // Replace extra quotes around txid string
+  if (!validateHash256(txid)) throw new Error(text);
+
+  return { txid } as TxBroadcastResultOk;
+}
+
+/**
+ * Lookup the nonce for an address from a core node
+ * @param opts.address - The Stacks address to look up the next nonce for
+ * @param opts.api - Optional API info (`.url` & `.fetch`) used for fetch call
+ * @return A promise that resolves to an integer
+ */
+export async function getNonce({
+  address,
+  api: apiOpt,
+}: {
+  /** The Stacks address to look up the next nonce for */
+  address: string;
+} & ApiParam): Promise<bigint> {
+  // todo: could derive the network from the address and use as default if no apiOd
+
+  const api = defaultApiLike(apiOpt);
+  const url = `${api.url}${ACCOUNT_PATH}/${address}?proof=0`;
+  const response = await api.fetch(url);
+
+  if (!response.ok) {
+    const msg = await response.text().catch(() => '');
+    throw new Error(
+      `Error fetching nonce. Response ${response.status}: ${response.statusText}. Attempted to fetch ${url} and failed with the message: "${msg}"`
```

It would feel a bit clunky as a user to get `failed with the message: ""` in the case that fetching the msg throws. Could we throw a different error if the msg is not available?

---

Nice thanks -- yeah not super pretty. This is the behavior live right now, this method got moved. But I'll open a new issue for it 🚀

---

Yep, works for me!